src/autoscaler/db/sqldb/binding_sqldb.go

```diff
@@ -50,43 +51,89 @@ func (bdb *BindingSQLDB) Close() error {
 	}
 
-	if rows.Next() {
-		var (
-			existingOrgId   string
-			existingSpaceId string
-		)
-		if err := rows.Scan(&existingOrgId, &existingSpaceId); err != nil {
-			bdb.logger.Error("create-service-instance", err, lager.Data{"query": query, "serviceinstanceid": serviceInstanceId, "orgid": orgId, "spaceid": spaceId})
-		}
-		rows.Close()
-		if existingOrgId == orgId && existingSpaceId == spaceId {
+	if existingInstance != nil {
+		if existingInstance.OrgId == serviceInstance.OrgId && existingInstance.SpaceId == serviceInstance.SpaceId && existingInstance.DefaultPolicy == serviceInstance.DefaultPolicy && existingInstance.DefaultPolicyGuid == serviceInstance.DefaultPolicyGuid {
 			return db.ErrAlreadyExists
 		} else {
 			return db.ErrConflict
 		}
 
 	}
-	rows.Close()
 
-	query = "INSERT INTO service_instance" +
-		"(service_instance_id, org_id, space_id) " +
-		" VALUES($1, $2, $3)"
-	_, err = bdb.sqldb.Exec(query, serviceInstanceId, orgId, spaceId)
+	query := "INSERT INTO service_instance" +
+		"(service_instance_id, org_id, space_id, default_policy, default_policy_guid) " +
+		" VALUES($1, $2, $3, $4, $5)"
+	_, err = bdb.sqldb.Exec(query, serviceInstance.ServiceInstanceId, serviceInstance.OrgId, serviceInstance.SpaceId, nullableString(serviceInstance.DefaultPolicy), nullableString(serviceInstance.DefaultPolicyGuid))
 
 	if err != nil {
-		bdb.logger.Error("create-service-instance", err, lager.Data{"query": query, "serviceinstanceid": serviceInstanceId, "orgid": orgId, "spaceid": spaceId})
+		bdb.logger.Error("create-service-instance", err, lager.Data{"query": query, "serviceinstance": serviceInstance})
 	}
 	return err
 }
 
+func (bdb *BindingSQLDB) GetServiceInstance(serviceInstanceId string) (*models.ServiceInstance, error) {
+	query := "SELECT org_id, space_id, default_policy, default_policy_guid FROM service_instance WHERE service_instance_id = $1"
+
+	rows, err := bdb.sqldb.Query(query, serviceInstanceId)
+	if err != nil {
+		bdb.logger.Error("get-service-instance", err, lager.Data{"query": query, "serviceinstanceid": serviceInstanceId})
+		return nil, err
+	}
+	defer rows.Close()
+
+	if rows.Next() {
+		var (
+			existingOrgId             string
+			existingSpaceId           string
+			existingDefaultPolicy     sql.NullString
+			existingDefaultPolicyGuid sql.NullString
+		)
+		if err := rows.Scan(&existingOrgId, &existingSpaceId, &existingDefaultPolicy, &existingDefaultPolicyGuid); err != nil {
+			bdb.logger.Error("get-default-policy", err, lager.Data{"query": query, "serviceinstanceid": serviceInstanceId})
+			return nil, err
+		}
+		return &models.ServiceInstance{
+			ServiceInstanceId: serviceInstanceId,
+			OrgId:             existingOrgId,
+			SpaceId:           existingSpaceId,
+			DefaultPolicy:     existingDefaultPolicy.String,
+			DefaultPolicyGuid: existingDefaultPolicyGuid.String,
+		}, nil
+	}
+	return nil, nil
+}
+
+func (bdb *BindingSQLDB) UpdateServiceInstance(serviceInstance models.ServiceInstance) error {
```

Comment:
missing UT for UpdateServiceInstance and GetAppIdsByInstanceId?

Reply:
Oops, sorry. Added.